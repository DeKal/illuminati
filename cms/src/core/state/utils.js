export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
