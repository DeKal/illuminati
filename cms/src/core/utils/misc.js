import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
