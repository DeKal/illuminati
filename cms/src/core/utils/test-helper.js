import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
