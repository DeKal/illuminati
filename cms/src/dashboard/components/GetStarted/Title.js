import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

const Title = withStyles(theme => ({
  root: {
    fontSize: theme.fontSizes.XXL,
    fontWeight: 600,
    padding: '9px 0 4px 0',
    lineHeight: 1.3
  }
}))(Typography)

export default Title
