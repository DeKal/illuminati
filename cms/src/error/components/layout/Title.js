import { styled } from '@material-ui/core/styles'

const Title = styled('h2')(({ theme }) => ({
  color: theme.colors.dark,
  fontSize: theme.fontSizes.XXXL,
  fontWeight: 700,
  textTransform: 'uppercase',
  margin: `${theme.spaces.xsmall}px`
}))

export default Title
