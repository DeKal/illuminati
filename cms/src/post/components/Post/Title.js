import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Title = ({ isEdited, title, editPostTitle }) => (
  <TextField
    label="Post title"
    required
    variant="outlined"
    value={title}
    multiline
    rowsMax={2}
    fullWidth
    error={isEdited}
    onChange={event => editPostTitle(event.target.value)}
  />
)

export default Title

Title.propTypes = {
  isEdited: PropTypes.bool,
  title: PropTypes.string,
  editPostTitle: PropTypes.func
}
