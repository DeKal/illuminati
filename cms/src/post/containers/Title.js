import { connect } from 'react-redux'
import { getPostTitle, getEditedTitle } from 'post/state/selectors'
import { editPostTitle } from 'post/state/actions'
import Title from 'post/components/Post/Title'

const mapStateToProps = state => {
  const editedTitle = getEditedTitle(state)
  const title = getPostTitle(state)
  return {
    title: editedTitle || '',
    isEdited: editedTitle !== title
  }
}

const mapDispatchToProps = {
  editPostTitle
}

export default connect(mapStateToProps, mapDispatchToProps)(Title)
