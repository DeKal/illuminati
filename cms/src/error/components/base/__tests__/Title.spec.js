import * as React from 'react'
import { mountWithTheme } from 'core/utils/test-helper'
import Title from '../Title'

describe('Title', () => {
  it('should render correctly', function() {
    const wrap = mountWithTheme(<Title />)
    expect(wrap).toMatchSnapshot()
  })
})
