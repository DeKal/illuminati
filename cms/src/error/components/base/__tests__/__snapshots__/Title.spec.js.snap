// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <Styled(h2)>
    <h2
      className="h2-root-1 h2-root-2"
    />
  </Styled(h2)>
</div>
`;
